src/core/client/admin/components/UserStatus/BanModal.tsx

```diff
@@ -1,76 +1,115 @@
-import { Localized } from "fluent-react/compat";
-import React, { FunctionComponent } from "react";
-
-import NotAvailable from "coral-admin/components/NotAvailable";
 import {
   Button,
   Card,
   CardCloseButton,
+  CheckBox,
   Flex,
   HorizontalGutter,
   Modal,
   Typography,
 } from "coral-ui/components";
+import { Localized } from "fluent-react/compat";
+
+import NotAvailable from "coral-admin/components/NotAvailable";
+import { Format, withFormat } from "coral-framework/lib/i18n";
+import React, { FunctionComponent, useState } from "react";
 
 import styles from "./BanModal.css";
 
 interface Props {
   username: string | null;
   open: boolean;
   onClose: () => void;
-  onConfirm: () => void;
+  onConfirm: (message?: string) => void;
+  format: Format;
 }
 
 const BanModal: FunctionComponent<Props> = ({
   open,
   onClose,
   onConfirm,
   username,
-}) => (
-  <Modal open={open} onClose={onClose} aria-labelledby="banModal-title">
-    {({ firstFocusableRef, lastFocusableRef }) => (
-      <Card className={styles.card}>
-        <CardCloseButton onClick={onClose} ref={firstFocusableRef} />
-        <HorizontalGutter size="double">
-          <HorizontalGutter>
-            <Localized
-              id="community-banModal-areYouSure"
-              strong={<strong />}
-              $username={username || <NotAvailable />}
-            >
-              <Typography variant="header2" id="banModal-title">
-                Are you sure you want to ban{" "}
-                <strong>{username || <NotAvailable />}</strong>?
-              </Typography>
-            </Localized>
-            <Localized id="community-banModal-consequence">
-              <Typography>
-                Once banned, this user will no longer be able to comment, use
-                reactions, or report comments.
-              </Typography>
-            </Localized>
-          </HorizontalGutter>
-          <Flex justifyContent="flex-end" itemGutter="half">
-            <Localized id="community-banModal-cancel">
-              <Button variant="outlined" onClick={onClose}>
-                Cancel
-              </Button>
-            </Localized>
-            <Localized id="community-banModal-banUser">
-              <Button
-                variant="filled"
-                color="primary"
-                onClick={onConfirm}
-                ref={lastFocusableRef}
+  format,
+}) => {
+  function getMessage(): string {
+    return format("community-banModal-emailTemplate", {
+      username,
+    });
+  }
+  const [emailMessage, setEmailMessage] = useState(getMessage());
```

Comment:
Use final form to track form changes instead of using `useState`. Refer to https://github.com/coralproject/talk/blob/ce2c7c0d8f44f5eb35b6240497e080f42911727b/src/core/client/admin/routes/Invite/InviteCompleteForm.tsx as an example.